### ISSUE TITLE
XSharp Visual Studio does not allow adding a property page, per project type guid, which depends on the build configuration

### ISSUE THREAD

**Issue by Adahel** [X-Sharp/XSharpPublic#992]
Good day!

**Describe the bug**
The project type guid property page is not displayed if it depends on the build configuration.

**To Reproduce**
Cannot reproduce the bug as I had the problem with a personal extension I was writing. I haven't tried it with other project type guides, but here's a list for anyone who wants to try it out: [List of Visual Studio Project Type GUIDs](https://www.codeproject.com/Reference/720512/List-of-Visual-Studio-Project-Type-GUIDs)

**Expected behavior**
The property page is displayed.

**Screenshots**
This image is not possible to reproduce without a hack extension. The "Content Build" property page is not displayed.
![XNA X#](https://user-images.githubusercontent.com/38345740/158060211-87490a47-3ac8-47de-be0c-9efb3b5864d2.JPG)
![XNA C#](https://user-images.githubusercontent.com/38345740/158073709-7d29ba96-be53-4ee9-af6f-9e2cdf130cfc.JPG)

**Desktop (please complete the following information):**
 - OS: Windows 10 21H2
 - Browser: Microsoft Edge (Chromium)
 - Version: 99.0.1150.36

**Smartphone (please complete the following information):**
I do not have, :(!

**Additional context**
I haven't tested it, but using a magic called debug, I believe the bug can be fixed by changing 1 line:
https://github.com/X-Sharp/XSharpPublic/blob/main/VisualStudio/ProjectBase/ProjectConfig.cs#L959
This line is different and does not contain the bug:
https://github.com/X-Sharp/XSharpPublic/blob/main/VisualStudio/ProjectBase/NodeProperties.cs#L388

Thank you very much for reading this issue.

**Comment by RobertvanderHulst** [comment]
Adahel,

Do you have an example of how you are using this ? And of what you are trying to achieve ?

Robert

**Comment by RobertvanderHulst** [comment]
Adahel,
I know how xsproj files work and I know all about the targets file. I wrote the code behind it.
Are you saying that when you add the `<XnaCompressContent>` property that a project property page should be added automatically

**Comment by Adahel** [comment]
Robert,
maybe I didn't express myself correctly and there is a communication problem, I apologize. I'm saying that the page that depends on "AnyCPU | x86 | x64" doesn't visible, because X#'s "VSHPROPID_CfgPropertyPagesCLSIDList" only takes the guids from X# itself. This is a bug, I'm not asking for a new feature. I don't understand if your comment was a question or a affirmation. Interestingly "VSHPROPID_PropertyPagesCLSIDList" was implemented correctly and does not suffer from this bug.

The bug is visible in the first image:

XNA Game Studio - OK. (VSHPROPID_PropertyPagesCLSIDList) "A768877C-2EF4-493D-9D74-572E9E894C00"
Application - OK. (VSHPROPID_PropertyPagesCLSIDList) "AA6C8D78-22FF-423A-9C7C-5F2393824E04"
Language - OK. (VSHPROPID_PropertyPagesCLSIDList) "AA6C8D78-22FF-423A-9C7C-5F2393824E04"
Dialect - OK. (VSHPROPID_PropertyPagesCLSIDList) "AA6C8D78-22FF-423A-9C7C-5F2393824E04"
Build - OK. (VSHPROPID_CfgPropertyPagesCLSIDList) "AA6C8D78-22FF-423A-9C7C-5F2393824E04"
Content Build - Not OK. (VSHPROPID_CfgPropertyPagesCLSIDList) "A768877C-2EF4-493D-9D74-572E9E894C00"
Build Events - OK. (VSHPROPID_CfgPropertyPagesCLSIDList) "AA6C8D78-22FF-423A-9C7C-5F2393824E04"
Debug. OK. (VSHPROPID_CfgPropertyPagesCLSIDList) "AA6C8D78-22FF-423A-9C7C-5F2393824E04"

**Comment by RobertvanderHulst** [comment]
Adahel,

Both code snippets are coming from the original MPF code from Microsoft.
I can change them to make them do the same.
However I would love to have an example so I can test to see that the code is doing what you expect it to do.
Without example I can change the code and hope (pray ?) that it does what you expect it to do.

To be 100% clear: I will not change the code until I have an example that shows how the extra XNA Game Studio GUID is added to the project system. 
The only difference that I see in both fragments is that one snippet calls GetProperty() on the projectnode itself and the other on the GetOuterHierarchy() of the projectnode. Is this the difference that you meant ?

Does this need an extension ? Where can I download that extension ?
Or is this activated in another way ?

Robert

**Comment by Adahel** [comment]
Modified (vsixmanifest) version of Microsoft XNA Game Studio : https://flatredball.com/visual-studio-2019-xna-setup/ (Original version is Visual Studio 2010).
X# adapter written by me: https://www.mediafire.com/file/4b4smy08evl44x9/Microsoft.Xna.XSharp.LanguageSupport.zip/file
(Notice: No templates temporarily)
Add:
```
<ProjectTypeGuids>{A768877C-2EF4-493D-9D74-572E9E894C00};{5DD37FEA-7D13-4164-ADE2-26D12C2BF5F6};{AA6C8D78-22FF-423A-9C7C-5F2393824E04}</ProjectTypeGuids>
<XnaFrameworkVersion>v4.0</XnaFrameworkVersion>
<XnaPlatform>Windows</XnaPlatform>
```
Proof that it's an X# bug:
F#:
![F#](https://user-images.githubusercontent.com/38345740/158217305-ea5f3cd6-bb34-4a10-b0b5-67efdbe2be66.JPG)
Nemerle:
![Nemerle](https://user-images.githubusercontent.com/38345740/158217851-f52cabe3-156e-48ae-a27b-3ff1c4186467.JPG)
Thank you and have a nice day.

**Comment by RobertvanderHulst** [comment]
Thank, I will have a look at this.
So this would allow you to write XNA games with X# ?
Do you have an example of this ?

**Comment by RobertvanderHulst** [comment]
I have adjusted the code. I am not going to test this to be honest. There are too many external dependencies and not enough customers that want this. 
I trust you will let me know if this works for you or not.
When you need more changes then please supply me the setup for the X# integration and the needed DLLs to activate XNA Studio.

**Comment by Adahel** [comment]
@RobertvanderHulst 
Mr. Robert Vander Hulst,
How thoughtful you were to me and trusted me to know if this works for me or not: I lost 1 whole day trying to compile the Visual Studio X# extension, trying to make pkgdef work: "every swear word imaginable for pkgdef"... Yes, it worked perfectly.
![XNA X#](https://user-images.githubusercontent.com/38345740/158511759-2fe147d8-2f52-466c-9fc7-dd9e6c88fec6.JPG)
 I opened this issue, because of all the issues I encountered, this was the one that couldn't be avoided with my code. I learned a lot about X# by looking at X# extension code written in X#, better than any documentation. Will X# 2.12 be released publicly after subscribers or will I have to wait for another release to receive this fix officially? My sincere thanks.
Átila César (Adahel).
